interp/src/primitives/combinational.rs

```diff
@@ -117,40 +274,132 @@ comb_primitive!(StdXor[WIDTH](left: WIDTH, right: WIDTH) -> (out: WIDTH) {
 
 // ===================== Comparison Operations ======================
 comb_primitive!(StdGt[WIDTH](left: WIDTH, right: WIDTH) -> (out: WIDTH) {
-  let left_64 = left.as_u64();
-  let right_64 = right.as_u64();
-  let init_val = left_64 > right_64;
-  Value::from_init(init_val, 1_usize).into()
+    let a_iter = left.vec.iter().by_ref();
+    let b_iter = right.vec.iter().by_ref();
+    let mut tr = false;
+
+    //as we proceed up in magnitude, it doesn't matter which port was less
+    //b/c [100....000] > [011....111] always.
+    //but if ai = bi, it matters which was higher previously
+    for (ai, bi) in a_iter.zip(b_iter) {
+        tr = ai & !bi || tr & !bi || tr & ai;
+    }
+
+    if tr {
+        Value::bit_high().into()
+    } else {
+        Value::bit_low().into()
+    }
 });
```

Comment:
That's valid, I didn't spend any time investigating the innate comparison between values. I just remember @yn224 commenting that Value's comparison was broken

Reply:
Yeah it's because it does a lexical comparison, but starting from the LSB to the MSB rather than the other way around, which doesn't produce the expected results